api/app/models/pox_calculator.rb

```diff
@@ -47,6 +47,44 @@ def self.calculate_usda_yields(names: ['stacker'], file: 'vaults-pox-33.json', u
     vaults
   end
 
+  def self.calculate_liqs
+    total_amount = 0
+    stacked_amount = 0
+    offset = 0
+    data = HTTParty.get("https://stacks-node-api.stacks.co/extended/v1/address/SP2C2YFP12AJZB4MABJBAJ55XECVS7E4PMMZ89YZR.arkadiko-auction-engine-v4-1/transactions?limit=50&offset=#{offset}&unanchored=true")
+    results = data.parsed_response['results']
+    while results.length > 0
+      results.each do |d|
+        txid = d['tx_id']
+        tx = HTTParty.get("https://stacks-node-api.mainnet.stacks.co/extended/v1/tx/#{txid}")
+        next if tx.parsed_response.nil? || tx.parsed_response['events'].length.zero?
+        begin
+          collateral_amount = tx.parsed_response['events'][0]['contract_log']['value']['repr'].split("(collateral u")[1].split(") ")[0]
+          stck_amount = tx.parsed_response['events'][0]['contract_log']['value']['repr'].split("(stacked-tokens u")[1].split(") ")[0]
+          total_amount += collateral_amount.to_i
+          stacked_amount += stck_amount.to_i
+        rescue
+          begin
+            collateral_amount = tx.parsed_response['events'][2]['contract_log']['value']['repr'].split("(collateral u")[1].split(") ")[0]
+            stck_amount = tx.parsed_response['events'][2]['contract_log']['value']['repr'].split("(stacked-tokens u")[1].split(") ")[0]
+            total_amount += collateral_amount.to_i
+            stacked_amount += stck_amount.to_i
+          rescue
+            puts "Failed TX with id #{txid}"
+          end
+        end
+        sleep(2)
+      end
+
+      puts "Total Collateral: #{total_amount}"
```

Comment:
Q: this also takes non stacking vaults into account? For these we never minted xSTX so they don't need to be included?

Reply:
This also takes non-stacking vaults into account, yes. However, this script was just written to have a rough calculation. I'll improve it either way to exclude non-stacking liquidated vaults.